src/votekit/outcome_model.py

```diff
@@ -0,0 +1,140 @@
+from pydantic import BaseModel
+from profile import PreferenceProfile
+from typing import Optional
+
+#1. "--> int" arrow thing for each function
+#2. is add_winners_and_losers necessary? --A: probabably not
+#3. do we want anything more?]
+#4. RCV cruncher: inspiration for funcitons--> statistics functions/things
+# Example of immutable data model for results
+
+class Outcome():
+    """
+    curr_round (an Int): current round number
+    elected (a set of Candidate): candidates who pass a certain threshold to win an election
+    eliminated (a set of Candidate): candidates who were eliminated (lost in the election)
+    remaining (a set of Candidate): candidates who are still in the running
+    rankings (a list of a set of Candidate): ranking of candidates with sets representing ties
+    ballots (a list of Dict): a list of ballot types (schedule preferences) and their frequency of times
+    winners_votes (a Dict of candidates and list of their ballots): each winner's list of ballots that elected them
+    """
+    # TODO (not andrew written): replace these with list/set[Candidate] once candidate is well-defined?
+    def __init__(self, curr_round: int, elected: list[str] = [], eliminated: list[str]=[], remaining: list[str]=[], winner_votes: dict={}, profile: PreferenceProfile=None,  previous =None):
+        self.curr_round = curr_round
+        self.elected = elected
+        self.eliminated = eliminated
+        self.remaining = remaining
+        self.profile = profile
+        self.winner_votes = winner_votes
+        self.previous = previous
+
+    class Config:
+        allow_mutation = False
+
+    def get_curr_round(self) -> int:
```

Comment:
there is no need to write accessors like this in Python, just have people access the variables directly if you're just returning them

Reply:
Deleted